### DISCUSSION TITLE
Use Tooltip on Docs

### DISCUSSION THREAD

**Discussion by shak2** [godotengine/godot-proposals#9923] [Editor]
Imagine hovering the mouse over the Documentations properties and methods and seeing the descriptions without having to scroll up and down and up and down then finsing where you were, then up and down again.

Is this possible to implement? Using HTML tooltips?

**Comment by AThousandShips** [comment]
The descriptions aren't guaranteed to be short, and there aren't any "brief descriptions" for them, this would easily just clutter things IMO, and I don't really see how it'd be any easier or better than just going to the description

**Comment by Mickeon** [comment]
It's not impossible (never say never), but no one has done it yet because it would require rewiring how the tooltips are generated in the docs (they currently just use the default Popup).
For most cases you actually don't want this anyway, as descriptions can be quite lengthy . But for constants and similar, I can see the benefit.

**Reply by AThousandShips** [reply]
Some other doc generation systems have short descriptions for each entry, which are often used in the table of contents and in tooltips

Without that the option would be to just grab a certain portion of the body and put it in a tooltip, but that feels unhelpful most of the time, as does putting the full description

Having a separate short description for members feels like a lot of extra effort, and risks just having unhelpful obvious descriptions if any, and adds work to decide the limit and discussion over when to use and not, etc.

**Reply by Mickeon** [reply]
I have been thinking of having the first newline of the description be worth something other than a generic break. Any text prior *could* serve as a short summary, as most docs tend to be written that way (at least, that's my style).
